src/app/GitUI/UserControls/FileStatusList.cs

```diff
@@ -572,274 +558,152 @@ public IEnumerable<FileStatusItem> FirstGroupItems
-                    if (item.Group == firstNonEmptyGroup && IsSearchableItem(item))
-                    {
-                        return i;
-                    }
-                }
-
-                return -1;
-            }
+            TreeNode? GetLastItem() => FileStatusListView.Items().LastOrDefault(IsSearchableItem);
 
-            int GetLastIndex()
-            {
-                if (FileStatusListView.Items.Count == 0)
-                {
-                    return -1;
-                }
-
-                if (FileStatusListView.Groups.Count < 2)
-                {
-                    return FileStatusListView.Items.Count - 1;
-                }
-
-                ListViewGroup? lastNonEmptyGroup = null;
-                for (int i = FileStatusListView.Groups.Count - 1; i >= 0; i--)
-                {
-                    if (ContainsSearchableItem(FileStatusListView.Groups[i]))
-                    {
-                        lastNonEmptyGroup = FileStatusListView.Groups[i];
-                        break;
-                    }
-                }
-
-                for (int i = FileStatusListView.Items.Count - 1; i >= 0; i--)
-                {
-                    ListViewItem item = FileStatusListView.Items[i];
-                    if (item.Group == lastNonEmptyGroup && IsSearchableItem(item))
-                    {
-                        return i;
-                    }
-                }
-
-                return -1;
-            }
-
-            static bool IsSearchableItem(ListViewItem item)
+            static bool IsSearchableItem(TreeNode item)
             {
                 return item.Tag is FileStatusItem fileStatusItem
                     && !fileStatusItem.Item.IsStatusOnly
                     && !fileStatusItem.Item.IsRangeDiff;
             }
-
-            static bool ContainsSearchableItem(ListViewGroup group)
-            {
-                foreach (ListViewItem item in group.Items)
-                {
-                    if (IsSearchableItem(item))
-                    {
-                        return true;
-                    }
-                }
-
-                return false;
-            }
         }
 
         public void SelectAll() => SelectItems(_ => true);
 
         public void SelectFirstVisibleItem()
         {
-            if (FileStatusListView.Items.Count == 0)
-            {
-                return;
-            }
+            FileStatusListView.SelectedNode = FileStatusListView.Items()
+                .FirstOrDefault(node => IsLeaf(node) && !IsCollapsed(node) && node.Tag is FileStatusItem fileStatusItem && fileStatusItem.Item != _noItemStatuses[0]);
+
+            return;
+
+            static bool IsLeaf(TreeNode node) => node.Nodes.Count == 0;
 
-            ListViewGroup? group = FileStatusListView.Groups().FirstOrDefault(gr => gr.Items.Count > 0);
-            if (group is not null)
+            static bool IsCollapsed(TreeNode node)
```

Comment:
I get possible null error (with .net9), also line 685
```suggestion
            static bool IsCollapsed(TreeNode? node)
```

Reply:
> I get possible null error (with .net9), also line 685

How can `node` be `null` in `FileStatusListView.Items().FirstOrDefault(node => ...)`?!?
I consider this a bug in .NET 9 if `#nullable enable` in `TreeViewExtensions.cs` does not help.

Reply:
It may be.
Use and merge is simpler if this is suppressed already in this PR...